pinot-common/src/main/java/com/linkedin/pinot/common/utils/helix/HelixHelper.java

```diff
@@ -95,13 +96,17 @@ public Boolean call() {
           BaseDataAccessor<ZNRecord> baseDataAccessor = dataAccessor.getBaseDataAccessor();
           boolean success;
 
+          // If the ideal state is large enough, enable compression
+          if (MAX_PARTITION_COUNT_IN_UNCOMPRESSED_IDEAL_STATE < updatedIdealState.getPartitionSet().size()) {
+            updatedIdealState.getRecord().setBooleanField("enableCompression", true);
+          }
+
           try {
             success =
                 baseDataAccessor.set(path, updatedIdealState.getRecord(), idealState.getRecord().getVersion(),
                     AccessOption.PERSISTENT);
           } catch (Exception e) {
-            LOGGER.warn("Caught exception: {} while updating ideal state for resource {}, retrying.", resourceName,
-                e.getMessage());
+            LOGGER.warn("Caught exception while updating ideal state for resource {}, retrying.", resourceName, e);
```

Comment:
I also thought about that when implementing this, but decided against it since we would log this message every time we'd update the ideal state, which might be too noisy and doesn't add much information:

- If we compress the znode, it'll show up as compressed in zooinspector
- If we don't and it exceeds the size, then we'll now get an exception that bubbles up properly as opposed to failing silently before